packages/@aws-cdk/core/lib/cfn-output.ts

```
@@ -115,6 +116,35 @@ export class CfnOutput extends CfnElement {
     this._exportName = exportName;
   }
 
+  /**
+   * Return the `Fn.importValue` expression to import this value into another stack
+   *
+   * The returned value should not be used in the same stack, but in a
```

Can we test that this is referenced in the same stack and fail early?